pinot-core/src/main/java/org/apache/pinot/core/operator/filter/RawValueInvertedIndexFilterOperator.java

```diff
@@ -0,0 +1,193 @@
+import org.apache.pinot.segment.local.segment.index.readers.RawValueBitmapInvertedIndexReader;
+import org.apache.pinot.segment.spi.datasource.DataSource;
+import org.apache.pinot.spi.data.FieldSpec.DataType;
+import org.roaringbitmap.buffer.ImmutableRoaringBitmap;
+import org.roaringbitmap.buffer.MutableRoaringBitmap;
+
+/**
+ * Filter operator that uses raw value bitmap inverted index to handle predicates on raw encoded columns.
+ */
+public class RawValueInvertedIndexFilterOperator extends BaseColumnFilterOperator {
+  private static final String EXPLAIN_NAME = "FILTER_RAW_INVERTED_INDEX";
+
+  private final PredicateEvaluator _predicateEvaluator;
+  private final RawValueBitmapInvertedIndexReader _invertedIndexReader;
+  private final DataType _dataType;
+  private final boolean _exclusive;
+
+  public RawValueInvertedIndexFilterOperator(QueryContext queryContext, PredicateEvaluator predicateEvaluator,
+      DataSource dataSource, int numDocs) {
+    super(queryContext, dataSource, numDocs);
+    _predicateEvaluator = predicateEvaluator;
+    _invertedIndexReader = (RawValueBitmapInvertedIndexReader) dataSource.getInvertedIndex();
+    _dataType = dataSource.getDataSourceMetadata().getDataType();
+    _exclusive = predicateEvaluator.isExclusive();
+  }
+
+  @Override
+  protected BlockDocIdSet getTrues() {
+    // Handle null predicate
+    if (_predicateEvaluator.isAlwaysFalse()) {
+      return EmptyDocIdSet.getInstance();
+    }
+    if (_predicateEvaluator.isAlwaysTrue()) {
+      return new BitmapDocIdSet(new MutableRoaringBitmap(), _numDocs);
+    }
+
+    // Get bitmap for each matching value and OR them together
+    MutableRoaringBitmap result = computeMatchingBitmap();
+
+
+    return new BitmapDocIdSet(result, _numDocs);
+  }
+
+  private MutableRoaringBitmap computeMatchingBitmap() {
+    MutableRoaringBitmap bitmap = new MutableRoaringBitmap();
+    Predicate predicate = _predicateEvaluator.getPredicate();
+    Predicate.Type predicateType = predicate.getType();
+    switch (predicateType) {
+      case EQ:
+        addMatchingValueBitmap(bitmap, ((EqPredicate) predicate).getValue());
+        break;
+      case NOT_EQ:
+        addMatchingValueBitmap(bitmap, ((NotEqPredicate) predicate).getValue());
+        break;
+      case IN:
+        for (String value : ((InPredicate) predicate).getValues()) {
+          addMatchingValueBitmap(bitmap, value);
+        }
+        break;
+      case NOT_IN:
+        for (String value : ((NotInPredicate) predicate).getValues()) {
+          addMatchingValueBitmap(bitmap, value);
+        }
+        break;
+      case RANGE:
+        // For range queries, we need to scan through all values and apply the predicate
+        // This is not efficient, but it's the only way to handle range queries with raw encoding
+        // TODO: Add support for range index with raw encoding
+        throw new UnsupportedOperationException("Range predicates not supported for raw encoded columns");
```

Comment:
Since dictionary is within the inverted index, I think we can add a method in `RawValueBitmapInvertedIndexReader` directly to fetch the bitmap by passing the range.